### ISSUE TITLE
Windows CSI Node DaemonSet is not Running/ErrImagePull

### ISSUE THREAD

**Issue by dschunack** [kubernetes-sigs/aws-ebs-csi-driver#1001]
/kind bug

**What happened?**
Windows CSI Node DaemonSet is not running due to an wrong Image without windows support for windows/amd64.

Looks for me that it was not tested on EKS Windows Nodes.

```
  Warning  Failed     15s                kubelet            Failed to pull image "k8s.gcr.io/provider-aws/aws-ebs-csi-driver:v1.1.3": rpc error: code = Unknown desc = no matching manifest for windows/amd64 10.0.17763 in the manifest list entries
```

**What you expected to happen?**
Running Windows CSI Node DaemonSet

**How to reproduce it (as minimally and precisely as possible)?**
Activate windows support in helm chart and spin up a Windows Node.

https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/charts/aws-ebs-csi-driver/values.yaml#L133

**Anything else we need to know?**:
```
Events:
  Type     Reason     Age                From               Message
  ----     ------     ----               ----               -------
  Normal   Scheduled  18s                default-scheduler  Successfully assigned kube-system/ebs-csi-node-windows-pt422 to ip-10-46-67-23.eu-central-1.compute.internal
  Normal   Pulling    16s                kubelet            Pulling image "k8s.gcr.io/provider-aws/aws-ebs-csi-driver:v1.1.3"
  Warning  Failed     15s                kubelet            Failed to pull image "k8s.gcr.io/provider-aws/aws-ebs-csi-driver:v1.1.3": rpc error: code = Unknown desc = no matching manifest for windows/amd64 10.0.17763 in the manifest list entries
  Normal   Pulled     15s                kubelet            Container image "k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.1.0" already present on machine
  Normal   Created    15s                kubelet            Created container node-driver-registrar
  Normal   Started    15s                kubelet            Started container node-driver-registrar
  Normal   Pulled     15s                kubelet            Container image "k8s.gcr.io/sig-storage/livenessprobe:v2.2.0" already present on machine
  Normal   Created    14s                kubelet            Created container liveness-probe
  Normal   Started    14s                kubelet            Started container liveness-probe
  Warning  Failed     11s (x2 over 12s)  kubelet            Error: ImagePullBackOff

```

**Environment**
- Kubernetes version (use `kubectl version`): 1.20
- Driver version: v1.1.3 
- Helm Chart v2.0.2

**Comment by wongma7** [comment]
Sorry, we checked in the node daemonset without actually releasing the image for it I added a disclaimer here that it's in pre-release state https://github.com/kubernetes-sigs/aws-ebs-csi-driver/tree/master/examples/kubernetes/windows#windows but ideally the chart shouldn't offer the option to install something that doesn't even exist yet.

https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/957 should fix it, after that I will make a release for 1.2.1 manifest list that contains windows image

/assign

**Comment by dschunack** [comment]
Hi, 

thanks for this info. We saw the option in the helm chart and we don't saw anything in the README.md that it is not finished yet. Now it's clear and we will wait on the new release.

**Comment by dschunack** [comment]
@wongma7 Hi, you released a new Container [image ](https://gallery.ecr.aws/b5w6x5z2/aws-ebs-csi-driver) that support Windows and Linux, but make it really sense to put a Linux and Windows Layer in a single Container Image that supports multi OS? 
We running currently more then 600 Linux EKS nodes(arm64/amd64) and 30 Windows Nodes. It makes absolute sense to support multi arch in the Container Image, but i think it's not an good idea to support multi OS in one Image, the overhead due to the different Container architecture is to large. 
All Linux nodes need to load the whole Container Image with the Windows Layer inside. That is a Hugh overhead per Node. As short example, 600 Linux Nodes x 2.5G for the Windows Layer = 1.5 TB overhead for the Windows Container Layer on Linux Nodes. It need also more Bandwidth, Storage and Time to load/start a new Container Image through a Update. 
I find it useful to split the images and to create a Linux and a Windows Container Image. It is also easier to support / integrate new features without affecting the other OS.
What do you think?

**Comment by wongma7** [comment]
@dschunack the container runtime (docker) should only pull the image that corresponds to the OS your container is running on.  even though on ECR or Docker Hub it says the image is 2+GB, your Linux Nodes won't bother to pull the 2 GB windows images, so there's no need to worry.

Basically now the tag is referring to a manifest list or "fat manifest", which links to multiple images. Docker knows how to pull the image for your OS and ignore the other ones. https://docs.docker.com/registry/spec/manifest-v2-2/

For reference https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/distribution/pull_v2_windows.go#L68

**Comment by iusergii** [comment]
@wongma7  It looks like 1.2.1 was released but still windows nodes can't pull the image `k8s.gcr.io/provider-aws/aws-ebs-csi-driver:v1.2.1`

**Comment by wongma7** [comment]
v1.3.0 in GCR (but not ECR) should have a windows image now.

**Comment by iusergii** [comment]
@wongma7 Updated my release with Helm Chart 2.2.1 and I can see Windows Image was successfully pulled on node:
```
REPOSITORY                                  TAG     IMAGE ID      CREATED     SIZE
k8s.gcr.io/provider-aws/aws-ebs-csi-driver  v1.3.0  93f2632e63a1  6 days ago  5.75GB
```

But Pod is Error/CrashLoopBackOff state.  **ebs-plugin** containers fails
```
I0923 17:21:23.071346   15268 metadata.go:101] retrieving instance data from ec2 metadata
I0923 17:21:23.317911   15268 metadata.go:108] ec2 metadata is available
panic: open \\.\\pipe\\csi-proxy-filesystem-v1: The system cannot find the file specified.

goroutine 1 [running]:
github.com/kubernetes-sigs/aws-ebs-csi-driver/pkg/driver.newNodeService(0xc000076a00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/kubernetes-sigs/aws-ebs-csi-driver/pkg/driver/node.go:93 +0x2b9
github.com/kubernetes-sigs/aws-ebs-csi-driver/pkg/driver.NewDriver(0xc00059ff40, 0x7, 0x7, 0x188603b0108, 0x100000000000011, 0xc000078000)
	/go/src/github.com/kubernetes-sigs/aws-ebs-csi-driver/pkg/driver/driver.go:94 +0x445
main.main()
	/go/src/github.com/kubernetes-sigs/aws-ebs-csi-driver/cmd/main.go:46 +0x257
```

**Comment by wongma7** [comment]
@iusergii is csi-proxy installed on the node? https://github.com/kubernetes-csi/csi-proxy check out https://github.com/kubernetes-sigs/aws-ebs-csi-driver/tree/master/examples/kubernetes/windows

**Comment by iusergii** [comment]
@wongma7  I haven't. Sorry, my bad.

**Comment by vmasule** [comment]
@wongma7 Thanks for speedy response, following is I see in the logs. I  hope this helps.

```
E1006 15:34:34.419805    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 15:36:36.546716    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 15:38:38.626608    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 15:40:40.663125    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 15:42:42.768905    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 15:44:44.857508    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 15:46:46.939181    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 15:48:48.984433    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 15:50:51.071841    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 15:52:53.203763    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 15:54:55.283390    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 15:56:57.338877    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 15:58:59.459646    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 16:01:01.574850    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 16:03:03.663959    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 16:05:05.792897    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 16:07:07.822685    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 16:09:09.868176    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 16:11:11.914355    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 16:13:13.983101    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 16:15:16.002326    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
E1006 16:17:18.062900    1360 driver.go:119] GRPC error: rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\d38a9777-330f-49ea-9316-a27cc295140a\\volumes\\kubernetes.io~csi\\pvc-671cb275-d9b8-4682-84dd-34212d7a6997\\mount": file does not exist
```

**Comment by vmasule** [comment]
@wongma7 One thing I forgot to update is that, I fallowed all the step except building `cis-proxy.exe` from source as it was resulting into error on my Apple M1, so I found the `exe` here https://k8scsi.blob.core.windows.net/csi-proxy/master/binaries/csi-proxy.tar.gz and installed it on Windows nodes. 

Is that version of CSI Proxy may be cause of above error?

**Comment by vmasule** [comment]
Thanks @wongma7, for your information I am using below helm chart

Repo: https://kubernetes-sigs.github.io/aws-ebs-csi-driver
Version: 2.3.0
image.repository: amazon/aws-ebs-csi-driver
image.tag: v1.3.0

I hope this help, apart from this I also have seen and just now reproduced below intermittent error 
```
Events:
  Type     Reason                Age                   From                                                                                      Message
  ----     ------                ----                  ----                                                                                      -------
  Warning  ProvisioningFailed    2m50s                 persistentvolume-controller                                                               storageclass.storage.k8s.io "windows" not found
  Normal   Provisioning          41s (x8 over 2m48s)   ebs.csi.aws.com_ebs-csi-controller-6cc677f5fb-km8xv_6c0725a7-b4eb-4ed9-95b3-fda4bf8b47a4  External provisioner is provisioning volume for claim "default/ebs-claim"
  Warning  ProvisioningFailed    41s (x8 over 2m48s)   ebs.csi.aws.com_ebs-csi-controller-6cc677f5fb-km8xv_6c0725a7-b4eb-4ed9-95b3-fda4bf8b47a4  failed to provision volume with StorageClass "windows": error generating accessibility requirements: no topology key found on CSINode ip-178-12-31-50.us-west-2.compute.internal
```

When I describe the node I can see those topology labels already present on that node.

**Comment by wongma7** [comment]
The node topology label issue sounds similar to https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/1030 , we have not solved the issue but can continue discussion there. 

Thanks for the information, I will comment again whe n I have results of my repro attempt

**Comment by vmasule** [comment]
Thank you @wongma7 , I must say speed with which fix provided really surprised me in open source community. Will be watching for release notification.

**Comment by wongma7** [comment]
it looks like a different issue, would it be possible to retrieve what the partitions on that volume look like? For example, if you can access powershell on that node the csi-proxy logs are from 

Get-Disk -Number 7 | Get-Partition
PartitionNumber  DriveLetter Offset                                        Size Type
---------------  ----------- ------                                        ---- ----
1                            17408                                     15.98 MB Reserved
2                            16777216                                   3.98 GB Basic

the workflow is slightly different between the first time the volume is used and second because for the first time the driver must format the volume but for the second time the driver must discover the partitions on the volume. So it is possible there is another bug in the discovery depending on how your disk looks.

**Comment by vmasule** [comment]
Actually, I terminated the node while ago but will reproduce it tomorrow and update here.

**Comment by vmasule** [comment]
Hi @wongma7, I reproduced what I promised yesterday, this what it looks like.

```
E1008 19:17:46.438765    3832 server.go:283] failed GetVolumeIDFromTargetPath: error getting the volume for the mount c:\var\lib\kubelet\plugins\kubernetes.io\csi\pv\pvc-4e8fe8a4-7e1a-453d-9eb6-90607d490f23\globalmount, internal error error getting volume from mount. cmd: (Get-Item -Path c:\var\lib\kubelet\plugins\ku
bernetes.io\csi\pv\pvc-4e8fe8a4-7e1a-453d-9eb6-90607d490f23\globalmount).Target, output: , error: <nil>
E1008 19:19:02.429445    3832 server.go:102] volume id empty
E1008 19:21:28.457506    3832 server.go:283] failed GetVolumeIDFromTargetPath: error getting the volume for the mount c:\var\lib\kubelet\plugins\kubernetes.io\csi\pv\pvc-4e8fe8a4-7e1a-453d-9eb6-90607d490f23\globalmount, internal error error getting volume from mount. cmd: (Get-Item -Path c:\var\lib\kubelet\plugins\ku
bernetes.io\csi\pv\pvc-4e8fe8a4-7e1a-453d-9eb6-90607d490f23\globalmount).Target, output: , error: <nil>
E1008 19:22:51.321570    3832 server.go:102] volume id empty
E1008 19:25:08.527540    3832 server.go:121] ListDiskIDs failed: The system cannot find the file specified.
E1008 19:27:36.302289    3832 server.go:283] failed GetVolumeIDFromTargetPath: error getting the volume for the mount c:\var\lib\kubelet\plugins\kubernetes.io\csi\pv\pvc-4e8fe8a4-7e1a-453d-9eb6-90607d490f23\globalmount, internal error error getting volume from mount. cmd: (Get-Item -Path c:\var\lib\kubelet\plugins\ku
bernetes.io\csi\pv\pvc-4e8fe8a4-7e1a-453d-9eb6-90607d490f23\globalmount).Target, output: , error: <nil>
E1008 19:29:03.477955    3832 server.go:102] volume id empty
E1008 19:31:13.038994    3832 server.go:283] failed GetVolumeIDFromTargetPath: error getting the volume for the mount c:\var\lib\kubelet\plugins\kubernetes.io\csi\pv\pvc-4e8fe8a4-7e1a-453d-9eb6-90607d490f23\globalmount, internal error error getting volume from mount. cmd: (Get-Item -Path c:\var\lib\kubelet\plugins\ku
bernetes.io\csi\pv\pvc-4e8fe8a4-7e1a-453d-9eb6-90607d490f23\globalmount).Target, output: , error: <nil>
PS C:\Windows\system32> ^C
PS C:\Windows\system32> Get-Disk -Number 7 | Get-Partition
PS C:\Windows\system32>
PS C:\Windows\system32> Get-Disk -Number 8 | Get-Partition


   DiskPath: \\?\scsi#disk&ven_msft&prod_virtual_disk#2&1f4adffe&0&000008#{53f56307-b6bf-11d0-94f2-00a0c91efb8b}

PartitionNumber  DriveLetter Offset                                                                                                                                           Size Type
---------------  ----------- ------                                                                                                                                           ---- ----
1                            1048576                                                                                                                                        128 MB Reserved
2                            135266304                                                                                                                                    19.87 GB Basic


PS C:\Windows\system32>

```

**Comment by vmasule** [comment]
I think it was same what I shared yesterday, except for disk number, which is instead 7 it is 8 now.

**Comment by wongma7** [comment]
OK, I thought there are 2 different errors so I would expect the logs to be different, but we can focus on the original one because that is from the latest version of the driver

### error 1 from driver v1.3.0 "file does not exist"
```
  Warning  FailedMount             112s (x13 over 12m)  kubelet                  MountVolume.SetUp failed for volume "pvc-462e8ac4-5ce3-4fde-9398-d0b799d1b128" : rpc error: code = Internal desc = Could not mount "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-462e8ac4-5ce3-4fde-9398-d0b799d1b128\\globalmount" at "c:\\var\\lib\\kubelet\\pods\\04a8aea3-fa7a-4165-8260-374a0269e2d6\\volumes\\kubernetes.io~csi\\pvc-462e8ac4-5ce3-4fde-9398-d0b799d1b128\\mount": file does not exist
```
for this error, I think it is fixed by https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/1081. If you are brave you may try the unstable/development version of the driver containing this fix at `gcr.io/k8s-staging-provider-aws/aws-ebs-csi-driver:v20211007-helm-chart-aws-ebs-csi-driver-2.3.0-12-g4d5d7e7f`, otherwise I plan to release the fix in v1.3.2 and will update this issue when it releases.

### error 2 from driver 8c6c7e0a590da44c635d629a0653dc73aaa5c9e4 "volume id empty"
```
 Warning  FailedMount  4m20s (x100 over 3h25m)  kubelet  MountVolume.MountDevice failed for volume "pvc-8b2a752d-4938-411f-9586-d629de7642ac" : rpc error: code = Internal desc = could not format "7" and mount it at "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-8b2a752d-4938-411f-9586-d629de7642ac\\globalmount": rpc error: code = Unknown desc = volume id empty
```
for this error, I may need CSI node logs to debug further, as the volume 8 looks fine to me from the above output so I am not sure how come the driver was not able to find it.

**Comment by vmasule** [comment]
@wongma7 , I can confirm the **Error 1** above is now fixed, I used the public docker repo instead of `gcr.io`. this is the `https://hub.docker.com/r/amazon/aws-ebs-csi-driver/tags?page=1&name=4d5d7e7fea2b4b1d178f43c79957c2bebc2b0721` tag I used which was built day you pushed changes on master.

**_But..._**

I have reproduced the **Error 2** again and below are the logs.

Events from Application Pod
```
Events:
  Type     Reason                  Age                    From                     Message
  ----     ------                  ----                   ----                     -------
  Normal   Scheduled               9m23s                  default-scheduler        Successfully assigned default/windows-server-iis-7c5fc8f6c5-h6k7t to ip-172-12-27-150.us-west-2.compute.internal
  Warning  FailedAttachVolume      9m23s                  attachdetach-controller  Multi-Attach error for volume "pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac" Volume is already exclusively attached to one node and can't be attached to another
  Normal   SuccessfulAttachVolume  3m15s                  attachdetach-controller  AttachVolume.Attach succeeded for volume "pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac"
  Warning  FailedMount             2m18s (x3 over 7m19s)  kubelet                  Unable to attach or mount volumes: unmounted volumes=[xpandion-vol-mount], unattached volumes=[xpandion-vol-mount]: timed out waiting for the condition
  Warning  FailedMount             53s (x3 over 2m16s)    kubelet                  MountVolume.MountDevice failed for volume "pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac" : rpc error: code = Internal desc = could not format "1" and mount it at "\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\\globalmount": rpc error: code = Unknown desc = volume id empty
``` 

Logs and partition details from CSI Node

```
PS C:\Windows\system32> Get-Content -Path "C:\csi-logs\csi-proxy.log" -Wait
Log file created at: 2021/10/08 12:29:57
Running on machine: EC2AMAZ-430EL3Q
Binary: Built with gc go1.17 for windows/amd64
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
I1008 12:29:57.456986    5964 main.go:124] Running as a Windows service.
I1008 12:29:57.489680    5964 main.go:54] Starting CSI-Proxy Server ...
I1008 12:29:57.489680    5964 main.go:55] Version: v1.0.1-12-gebdbe95
I1008 12:29:57.457986    5964 main.go:134] Windows Service initialized through SCM
E1011 16:55:15.312005    5964 server.go:283] failed GetVolumeIDFromTargetPath: error getting the volume for the mount c:\var\lib\kubelet\plugins\kubernetes.io\csi\pv\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\globalmount, internal error error getting volume from mount. cmd: (Get-Item -Path c:\var\lib\kubelet\plugins\ku
bernetes.io\csi\pv\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\globalmount).Target, output: , error: <nil>
E1011 16:55:23.257404    5964 server.go:102] volume id empty
E1011 16:55:27.533375    5964 server.go:283] failed GetVolumeIDFromTargetPath: error getting the volume for the mount c:\var\lib\kubelet\plugins\kubernetes.io\csi\pv\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\globalmount, internal error error getting volume from mount. cmd: (Get-Item -Path c:\var\lib\kubelet\plugins\ku
bernetes.io\csi\pv\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\globalmount).Target, output: , error: <nil>
E1011 16:55:36.835518    5964 server.go:102] volume id empty
E1011 16:55:41.464118    5964 server.go:283] failed GetVolumeIDFromTargetPath: error getting the volume for the mount c:\var\lib\kubelet\plugins\kubernetes.io\csi\pv\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\globalmount, internal error error getting volume from mount. cmd: (Get-Item -Path c:\var\lib\kubelet\plugins\ku
bernetes.io\csi\pv\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\globalmount).Target, output: , error: <nil>
E1011 16:56:46.493897    5964 server.go:102] volume id empty
E1011 16:57:25.287261    5964 server.go:283] failed GetVolumeIDFromTargetPath: error getting the volume for the mount c:\var\lib\kubelet\plugins\kubernetes.io\csi\pv\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\globalmount, internal error error getting volume from mount. cmd: (Get-Item -Path c:\var\lib\kubelet\plugins\ku
bernetes.io\csi\pv\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\globalmount).Target, output: , error: <nil>
E1011 16:58:56.548325    5964 server.go:102] volume id empty
E1011 16:59:48.920536    5964 server.go:283] failed GetVolumeIDFromTargetPath: error getting the volume for the mount c:\var\lib\kubelet\plugins\kubernetes.io\csi\pv\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\globalmount, internal error error getting volume from mount. cmd: (Get-Item -Path c:\var\lib\kubelet\plugins\ku
bernetes.io\csi\pv\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\globalmount).Target, output: , error: <nil>
E1011 17:00:15.145688    5964 server.go:102] volume id empty
E1011 17:00:35.049874    5964 server.go:283] failed GetVolumeIDFromTargetPath: error getting the volume for the mount c:\var\lib\kubelet\plugins\kubernetes.io\csi\pv\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\globalmount, internal error error getting volume from mount. cmd: (Get-Item -Path c:\var\lib\kubelet\plugins\ku
bernetes.io\csi\pv\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\globalmount).Target, output: , error: <nil>
E1011 17:00:42.966245    5964 server.go:102] volume id empty
E1011 17:01:51.511164    5964 server.go:283] failed GetVolumeIDFromTargetPath: error getting the volume for the mount c:\var\lib\kubelet\plugins\kubernetes.io\csi\pv\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\globalmount, internal error error getting volume from mount. cmd: (Get-Item -Path c:\var\lib\kubelet\plugins\ku
bernetes.io\csi\pv\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\globalmount).Target, output: , error: <nil>
E1011 17:03:34.006764    5964 server.go:102] volume id empty
E1011 17:04:53.555044    5964 server.go:283] failed GetVolumeIDFromTargetPath: error getting the volume for the mount c:\var\lib\kubelet\plugins\kubernetes.io\csi\pv\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\globalmount, internal error error getting volume from mount. cmd: (Get-Item -Path c:\var\lib\kubelet\plugins\ku
bernetes.io\csi\pv\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\globalmount).Target, output: , error: <nil>
E1011 17:05:12.960786    5964 server.go:102] volume id empty
E1011 17:07:52.958427    5964 server.go:283] failed GetVolumeIDFromTargetPath: error getting the volume for the mount c:\var\lib\kubelet\plugins\kubernetes.io\csi\pv\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\globalmount, internal error error getting volume from mount. cmd: (Get-Item -Path c:\var\lib\kubelet\plugins\ku
bernetes.io\csi\pv\pvc-1c0ecde3-cb0d-49a6-8e68-156f1738ceac\globalmount).Target, output: , error: <nil>
E1011 17:09:19.112653    5964 server.go:102] volume id empty

PS C:\Windows\system32> Get-Disk -Number 1 | Get-Partition


   DiskPath: \\?\scsi#disk&ven_aws&prod_pvdisk#003400#{53f56307-b6bf-11d0-94f2-00a0c91efb8b}

PartitionNumber  DriveLetter Offset                                                                                                                                           Size Type
---------------  ----------- ------                                                                                                                                           ---- ----
1                            17408                                                                                                                                        15.98 MB Reserved
2                            16777216                                                                                                                                     49.98 GB Basic

```

Let me know if you need anything else. Thanks for the support.
